pkg/provider/validators.go

```diff
@@ -60,3 +61,14 @@ func validateUUID(v interface{}, _ cty.Path) diag.Diagnostics {
 	}
 	return nil
 }
+
+func startDayValidator() validator.Float64 {
+	return float64validator.Between(0, 6)
+}
+
+func startTimeValidator() validator.String {
+	return stringvalidator.RegexMatches(
+		regexp.MustCompile(`^(?:[01]\d|2[0-3]):[0-5]\d$`),
```

Comment:
```suggestion
		regexp.MustCompile(`^([01]\d|2[0-3]):[0-5]\d$`),
```
We probably don't need the non-capturing group in the regex. It would save minimal space, but I would prefer not to use them to increase the readability of the regex. 

If there is any advantage of using them, and I'm not seeing it, I'd love to learn.

Reply:
sure maybe i got the regex a bit wrong

Reply:
Nope, your regex is correct. When a group is found, regex captures it in the memory. The non-capturing group doesn't capture it in the memory, just interested in if the regex matches or not. As we're not processing the captured groups, it makes sense to use the non-capturing groups, but it just makes the regex harder to read. 🤷 
That was what I meant in the above comment. 
Other than that, both regex strings work as expected.

Reply:
https://manifold.net/doc/radian/why_do_non-capture_groups_exist_.htm